packages/firebase_crashlytics/lib/src/firebase_crashlytics.dart

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+part of firebase_crashlytics;
+
+/// The entry point for accessing Crashlytics.
+///
+/// You can get an instance by calling `Crashlytics.instance`.
+class Crashlytics {
+  static final Crashlytics instance = Crashlytics();
+
+  /// Set to true to have Errors sent to Crashlytics while in debug mode.
+  bool reportInDevMode = false;
+
+  /// Keys to be included with report.
+  @visibleForTesting
+  final Map<String, dynamic> keys = <String, dynamic>{};
+
+  /// Logs to be included with report.
+  @visibleForTesting
+  final ListQueue<String> logs = ListQueue<String>(15);
+  int logSize = 0;
+
+  bool get isInDebugMode {
+    bool _inDebugMode = false;
+    if (!reportInDevMode) {
+      assert(_inDebugMode = true);
+    }
+    return _inDebugMode;
+  }
+
+  static const MethodChannel channel =
+      MethodChannel('plugins.flutter.io/firebase_crashlytics');
+
+  Future<void> onError(FlutterErrorDetails details) async {
+    print('Error caught by Crashlytics plugin:');
+    if (isInDebugMode && !reportInDevMode) {
+      print(Trace.format(details.stack).trimRight().split('\n'));
+    } else {
+      // Send logs
+      await sendLogs();
+
+      // Send keys
+      await sendKeys();
+
+      // Report error
+      final List<String> stackTraceLines =
+          Trace.format(details.stack).trimRight().split('\n');
+      final dynamic result =
+          // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+          // https://github.com/flutter/flutter/issues/26431
+          // ignore: strong_mode_implicit_dynamic_method
+          await channel.invokeMethod('Crashlytics#onError', <String, dynamic>{
+        'exception': details.exceptionAsString(),
+        'stackTrace': details.stack.toString(),
+        'stackTraceLines': stackTraceLines,
+        'code': stackTraceLines[0].hashCode
+      });
+      print(result);
+    }
+  }
+
+  void crash() {
+    throw StateError('Error thrown by Crashlytics plugin');
+  }
+
+  Future<bool> isDebuggable() async {
```

Comment:
Needs dartdoc. Clarify that this means debuggability of the OEM parts of the app, not the Flutter app.

Reply:
Used doc from Android, but will have to dig a bit deeper to get more info on what this means in both a Android and iOS context.